plugin/inlay_hint.py

```diff
@@ -142,12 +145,21 @@ def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uui
             })
             result += f'<a href="{inlay_hint_click_command}">'
         instruction_text = '\nDouble-click to insert' if has_text_edits else ""
-        result += f'<span title="{(tooltip + instruction_text).strip()}">{html.escape(label)}</span>'
+        truncated = len(label) >= truncate_limit and truncate_limit > 0
+        truncated_label = label[:truncate_limit] + '…' if truncated else label
+        result_tooltip = (tooltip + instruction_text).strip()
+        result += f'<span title="{result_tooltip}">{html.escape(truncated_label)}</span>'
         if is_clickable:
             result += "</a>"
+        if truncated:
+            truncation_tooltip = html.escape(label)
+            result = f"<span title=\"{truncation_tooltip}\">{result}</span>"
```

Comment:
I have not tested it but this looks like it would result in something like
```html
<span title="Truncation tooltip"><span title="Regular tooltip">Label</span></span>
```
I don't think this would work, i.e. it would only show `Truncation tooltip` on hover, instead of both of them, or does it?

Reply:
You are right, but it works as expected. See comment: https://github.com/sublimelsp/LSP/pull/2514#issuecomment-2379405302

Reply:
Then let's first clarify what is expected. In your example "test" would be the full `label` of the [InlayHint](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#inlayHint), and "this is subspan?" would be the (optional) `tooltip`. I would expect that the `tooltip` is always visible when hovering (if it exists), and the full `label` is additionally visible if it is truncated. Your screenshot suggests that `tooltip` would be hidden if the label gets truncated.

What was wrong with only having a single `title` attribute and appending/prepending the `truncation_tooltip` content plus linebreak?

(Besides that, it seems the `tooltip` can be [MarkupContent](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markupContentInnerDefinition), so I'm not sure whether the assumption from https://github.com/sublimelsp/LSP/pull/2514#issuecomment-2376144994 is correct. But in this case we're probably processing it not quite correctly anyways, which is a different issue.)

Reply:
> visible

Sorry, I missed your point earlier. Now I understand. The label for truncated content won't be shown if a label_part has a tooltip, but it will be displayed in areas where the tooltip is missing. I didn't see it as an issue, but I get your concern now.

For appending/prepending the tooltip, I'm not sure yet. One idea is to append the tooltip on a new line with the full label. Does that sound good?

Reply:
Yes, this was my initial idea. I'm not aware of any server which uses the optional `tooltip`, so perhaps we cannot test it directly. But it looks like you already had implemented it in this way before https://github.com/sublimelsp/LSP/pull/2514/commits/0fcdf798ecc1a97dd7bf44b4275d317a2f1548f5. Didn't it work correctly?

Reply:
> I have not tested it but this looks like it would result in something like
> 
> ```
> <span title="Truncation tooltip"><span title="Regular tooltip">Label</span></span>
> ```
> 
> I don't think this would work, i.e. it would only show `Truncation tooltip` on hover, instead of both of them, or does it?

As far as I remember, this will show the "Regular tooltip" when hovering `Label`.

Whatever we do, it won't be very apparent to the user where each tooltip starts and ends unless we maybe add some additional styling.

Reply:
> Yes, this was my initial idea. I'm not aware of any server which uses the optional `tooltip`, so perhaps we cannot test it directly. But it looks like you already had implemented it in this way before [0fcdf79](https://github.com/sublimelsp/LSP/commit/0fcdf798ecc1a97dd7bf44b4275d317a2f1548f5). Didn't it work correctly?

I thought that wrapping with `<span>' would be more appropriate. I guess I'll implement with append/prepend.

Reply:
Done in [c6e0055](https://github.com/sublimelsp/LSP/pull/2514/commits/c6e0055b849610f937968c7ed8c80b6fb599124f)